### PR TITLE
fix(argo-rollouts): Add missing liveness and readiness probe

### PR DESCRIPTION
**Pull request by huikang**
Signed-off-by: Hui Kang <hui.kang@salesforce.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [ ] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by huikang** [comment]
Refer to https://github.com/argoproj/argo-rollouts/blob/cf19b5b88089adcf4cc0b493de2db232cfb9181c/manifests/base/argo-rollouts-deployment.yaml#L26-L42

**Comment by huikang** [comment]
Hi, @mkilchhofer , could you help review this PR or shall I ping someone else? Thanks.

**Comment by mkilchhofer** [comment]
> Hi, @mkilchhofer , could you help review this PR or shall I ping someone else? Thanks.

I do not use the rollouts project, so I can only support with helm related stuff here. But I think I'd move the full liveness/readiness probes to values like this:
```yaml
          {{- if .Values.livenessProbe.enabled }}
          livenessProbe:
            {{- omit .Values.livenessProbe "enabled" | toYaml | nindent 12 }}
          {{- end }}
          {{- if .Values.readinessProbe.enabled }}
          readinessProbe:
            {{- omit .Values.readinessProbe "enabled" | toYaml | nindent 12 }}
          {{- end }}
```
WDYT?

**Comment by huikang** [comment]
Hi, @mkilchhofer , thanks for reviewing. The PR is updated as you suggested, except for always enabling the liveness and readiness probe which I believe are desired feature like in other projects, e.g., argo-cd deployment.

File: charts/argo-rollouts/templates/argo-rollouts-deployment.yaml [hunk 1]
```diff
@@ -39,6 +39,10 @@ spec:
         ports:
         - containerPort: 8090
           name: metrics
+        livenessProbe:
+          {{- .Values.controller.livenessProbe | toYaml | nindent 10 }}
```
Comment (mkilchhofer) [inline charts/argo-rollouts/templates/argo-rollouts-deployment.yaml:43]:
```suggestion
          {{- toYaml .Values.controller.livenessProbe | nindent 10 }}
```

File: charts/argo-rollouts/templates/argo-rollouts-deployment.yaml [hunk 2]
```diff
@@ -39,6 +39,10 @@ spec:
...
+        readinessProbe:
+          {{- .Values.controller.readinessProbe | toYaml | nindent 10 }}
```
Comment (mkilchhofer) [inline charts/argo-rollouts/templates/argo-rollouts-deployment.yaml:45]:
```suggestion
          {{- toYaml .Values.controller.readinessProbe | nindent 10 }}
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

**Review by mkilchhofer — APPROVED** [review]

File: charts/argo-rollouts/Chart.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: "v1.0.2"
 description: A Helm chart for Argo Rollouts
 name: argo-rollouts
-version: 1.0.3
+version: 1.0.4
 icon: https://raw.githubusercontent.com/argoproj/argo/master/argo.png
```
Comment (mkilchhofer) [inline charts/argo-rollouts/Chart.yaml:6]:
```suggestion
icon: https://argoproj.github.io/argo-rollouts/assets/logo.png
```

File: charts/argo-rollouts/Chart.yaml [hunk 4]
```diff
@@ -11,4 +11,4 @@ maintainers:
   - name: jessesuen
 annotations:
   artifacthub.io/changes: |
-    - "[Added]: Initialize Changelog"
+    - "[Fixed]: Add missing liveness and readiness probes"
```
Comment (mkilchhofer) [inline charts/argo-rollouts/Chart.yaml:14]:
```suggestion
    - "[Fixed]: Add missing liveness and readiness probes"
    - "[Changed]: Fix icon url"
```

**Pull request merged**